core/server/adapters/scheduling/SchedulingDefault.js

```diff
@@ -197,61 +197,61 @@ SchedulingDefault.prototype._execute = function (jobs) {
  * - if we detect to publish a post in the past (case blog is down), we add a force flag
  */
 SchedulingDefault.prototype._pingUrl = function (object) {
-    debug('Ping url', object.url, moment().format('YYYY-MM-DD HH:mm:ss'), moment(object.time).format('YYYY-MM-DD HH:mm:ss'));
+    const {url, time} = object;
+
+    debug('Ping url', url, moment().format('YYYY-MM-DD HH:mm:ss'), moment(time).format('YYYY-MM-DD HH:mm:ss'));
 
     let timeout;
-    const {url, time} = object;
-    const httpMethod = object.extra ? object.extra.httpMethod : 'PUT',
-        tries = object.tries || 0,
-        requestTimeout = object.extra ? object.extra.timeoutInMS : 1000 * 5,
-        maxTries = 30,
-        req = request[httpMethod.toLowerCase()](url),
-        self = this;
+    const httpMethod = object.extra ? object.extra.httpMethod : 'PUT';
+    const tries = object.tries || 0;
+    const requestTimeout = object.extra ? object.extra.timeoutInMS : 1000 * 5;
+    const maxTries = 30;
+
+    const options = {
+        timeout: requestTimeout,
+        method: httpMethod.toLowerCase(),
+        json: true
+    };
 
     if (moment(time).isBefore(moment())) {
         if (httpMethod === 'GET') {
-            req.query('force=true');
+            // @todo: rename to searchParams when updating to Got v10
+            options.query = 'force=true';
         } else {
-            req.send({
-                force: true
-            });
+            options.body = {force: true};
         }
     }
 
-    req.timeout({
-        response: requestTimeout
-    });
-
-    req.end(function (err, response) {
-        if (err) {
-            // CASE: post/page was deleted already
-            if (response && response.status === 404) {
-                return;
-            }
-
-            // CASE: blog is in maintenance mode, retry
-            if (response && response.status === 503 && tries < maxTries) {
-                timeout = setTimeout(function pingAgain() {
-                    clearTimeout(timeout);
+    return request(url, options).catch((err) => {
```

Comment:
It's possible that the error isn't an HTTP error - for example, in the test suite, the server response with `OK` as the body. Since Got is instructed to send and receive JSON (line 213), a parse error is thrown. Should there be a more explicit check for a parse error? I don't think it's necessary because the default functionality is to log it.

Reply:
We ended up using a mix of `json:true` (in oembed) and `headers['Content-Type'] = 'application/json';` (all the other usages of `lib/request'`). I think, to be on the safe side we should stick to `headers['Content-Type'] = 'application/json';` here as well, because scheduling cares only about the status code in the response and should not fail if the returned body is not JSON (in case somebody implemented their own publishing endpoint). 

@kirrg001 this is an interesting one to think about, should we be more liberal with returned content (no `accept:"application/json"` header) or only accept JSON in other usecases as well (slack ping, webhooks, update-check).